### PR TITLE
TIP_121c (cont): Aave Companies veBAL OTC, deprecate Tribe DAO sub-systems and contracts

### PR DESCRIPTION
**Pull request by eswak**
# TIP_121c (cont): Aave Companies veBAL OTC, deprecate Tribe DAO sub-systems and contracts

## Summary
Performs a veBAL OTC with Aave Companies and deprecates various Tribe DAO smart contracts, including: pcvGuardian, pcvSentinel and the ratioPCVControllerV2. 

Also removes various empty deposits from the Collaterisation Oracle.

## veBAL OTC spec
Add a `VeBalHelper` contract that will receive the appropriate roles from Tribe DAO in order to allow an external entity (OTC buyer) to manage the veBAL position currently held by the TRIBE DAO (112k veBAL).

Upgrade the implementation of the proxy named `balancerGaugeStaker` to a `BalancerGaugeStakerV2` contract, which allows to manage boost delegations, and give this proxy admin to Aave Companies multisig.

At the end of proposal execution, the contract states is the following : 

- `VeBalDelegatorPCVDeposit` is the contract who has the 112k veBAL lock
- `BalancerGaugeStaker` (proxy) is the contract that has the hardcoded delegated right to manage `VeBalDelegatorPCVDeposit`'s boosts
- `Aave Companies` --> owner of -> `VeBalHelper`
- `Aave Companies` --> proxy admin of -> `BalancerGaugeStaker`
- `VeBalHelper` --> owner of -> `BalancerGaugeStaker` (proxy)
- `VeBalHelper` --> Has Fei Roles
  - `PCV_CONTROLLER_ROLE`
  - `METAGOVERNANCE_TOKEN_STAKING`
  - `METAGOVERNANCE_GAUGE_ADMIN`
  - `METAGOVERNANCE_VOTE_ADMIN`
- `VeBalHelper` --> Can call `BalancerGaugeStaker` (proxy) and `VeBalDelegatorPCVDeposit`

Addresses:

- `Aave Companies` -> `0xe8d4D93D9728bD673b0197673a230F62255C7846`
- `VeBalDelegatorPCVDeposit` -> `0xc4EAc760C2C631eE0b064E39888b89158ff808B2`
- `BalancerGaugeStaker` (proxy) -> `0x66977Ce30049CD0e443216Bf26377966c3A109E2`
- `BalancerGaugeStaker` (implementation V2) -> `0xbf3f6c7821E4d7D5DF204dEa4871b2e8D68c1CCd`
- `VeBalHelper` -> `0xb02CE4D72124b98Df4EAB4184467d7Da0023F9FB`

Checklist: 

- [x] All Tests Passing
- [x] Proposal Added to ProposalsConfig
- [x] Fork Block Correct
- [x] Remove Any .only's on Tests
- [ ] Deploy Contract If Needed (+ Verify on Etherscan & Add to MainnetAddresses)
- [x] Update Documentation If Needed
- [x] Update Roles Config
- [x] Update Collateralization Oracle If Needed
- [ ] Proposal Submitted

### CODE REVIEW

File: contracts/metagov/VeBalHelper.sol [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    // OTC Management
+    // ----------------------------------------------------------------------------------
+
+    /// @notice perform OTC
+    function otcBuy() external {
+        require(msg.sender == otcBuyer, "UNAUTHORIZED");
+        otcToken.safeTransferFrom(msg.sender, otcDestination, otcAmount);
+        _transferOwnership(msg.sender);
+        BOOST_RECIPIENT_PROXY.changeAdmin(msg.sender);
+    }
```
Comment (eswak) [inline contracts/metagov/VeBalHelper.sol]: Potentially need to update this if the OTC is performed in a less trustless way (Aave suggested holding DAI in a multisig separately from the ownership transfer)

File: proposals/dao/tip_vebalotc.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+import { BigNumber } from 'ethers';
+import { getImpersonatedSigner, overwriteChainlinkAggregator, time } from '@test/helpers';
+import { TransactionResponse } from '@ethersproject/providers';
+import { forceEth } from '@test/integration/setup/utils';
+
+const PROXY_ADMIN_STORAGE_SLOT = '0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103';
+
+const fipNumber = 'vebalotc';
+
+const vebalOtcBuyer = '0x6ef71cA9cD708883E129559F5edBFb9d9D5C6148';
```
Comment (eswak) [inline proposals/dao/tip_vebalotc.ts]: To be updated with the address of the buyer

File: proposals/description/tip_vebalotc.ts [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  This proposal sets up the protocol to be able to transfer its 112k veBAL to Aave Companies for OTC trade.
+
+  This proposal also includes several cleanup items to wrap up previous proposals.
+
+  The full list of actions of this proposal is :
+  - 1. Handle veBAL OTC
+  - 2. Move dust Balancer-related funds to TC Multisig for selling
+  - 3. Cleanup LUSD->DAI Auction contracts (send dust to TC multisig for selling)
+  - 4. Cleanup WETH->DAI Auction contracts (send dust to TC multisig for selling)
+  - 5. Cleanup FEI/TRIBE on TC Timelock from the Rari Infra team clawback (send TRIBE to DAO treasury, burn FEI)
```
Comment (eswak) [inline proposals/description/tip_vebalotc.ts]: These might need removal if another proposal passes onchain before & takes care of these cleanup items.

Comment (Joeysantoro) [reply proposals/description/tip_vebalotc.ts]: Imo the cleanup items should be a separate PR

File: proposals/description/tip_vebalotc.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
+import { ethers } from 'ethers';
```
Comment (thomas-waite) [inline proposals/description/tip_vebalotc.ts]: You've removed an e2e test - do you also need to reduce the number of CI runners on the CircleCI config? If there's more runners/workers than tests, then the last one kind of "wraps around" and runs the entire set of e2es itself and takes ages

File: contracts/metagov/VeBalHelper.sol [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+pragma solidity ^0.8.0;
+
+import {VeBalDelegatorPCVDeposit} from "./VeBalDelegatorPCVDeposit.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {SafeERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {TransparentUpgradeableProxy, ProxyAdmin} from "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+
+/// @title Helper contract for veBAL OTC
+/// @author eswak
+contract VeBalHelper is Ownable {
```
Comment (thomas-waite) [inline contracts/metagov/VeBalHelper.sol:12]: To make sure I follow, we can't just transfer the `veBAL`. Instead we need to create this wrapper contract which the OTC buyer will use to effectively assume ownership - and it exposes the various needed methods like `lock()`, `setGaugeController()` etc which are decoupled from Tribe's access permissions

File: proposals/dao/tip_vebalotc.ts [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+const vebalOtcAmount = ethers.utils.parseEther('1000000'); // 1M
```
Comment (thomas-waite) [inline proposals/dao/tip_vebalotc.ts]: Personal preference but prefer `ethers.constants.WeiPerEther.mul(1_000_00)` syntax for these so can easily see order of magnitude

File: contracts/metagov/VeBalHelper.sol (hunk 5 above)
Comment (eswak) [reply contracts/metagov/VeBalHelper.sol:12]: 💯 correct

File: proposals/dao/tip_vebalotc.ts (hunk 6 above)
Comment (eswak) [reply proposals/dao/tip_vebalotc.ts]: ok

File: proposals/description/tip_vebalotc.ts (hunk 4 above)
Comment (eswak) [reply proposals/description/tip_vebalotc.ts]: yes good point

File: proposals/description/tip_vebalotc.ts (hunk 3 above)
Comment (eswak) [reply proposals/description/tip_vebalotc.ts]: ok

Comment (eswak) [reply proposals/description/tip_vebalotc.ts]: I've split into a separate proposal file, we'll want to consolidate it with the next vote that goes onchain

File: proposals/description/tip_vebalotc.ts (hunk 4 above)
Comment (eswak) [reply proposals/description/tip_vebalotc.ts:1]: ha good call

**Comment by eswak** [comment]
Changed PR base to execute after https://github.com/fei-protocol/fei-protocol-core/pull/1031

File: contracts/metagov/VeBalHelper.sol [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    /// @notice Withdraw ERC20 tokens that are on the pcvDeposit
+    /// @dev this will be needed to withdraw B-80BAL-20WETH after exitLock(),
+    /// but also to withdraw BAL and bb-a-usd earned from protocol fees
+    function withdrawERC20(
+        address token,
+        address to,
+        uint256 amount
+    ) external onlyOwner {
+        pcvDeposit.withdrawERC20(token, to, amount);
+    }
```
Comment (eswak) [inline contracts/metagov/VeBalHelper.sol:175]: Joey: restrict `token` to `bb-a-usd`, `BAL`, and `B-80BAL-20WETH`

File: proposals/description/tip_vebalotc.ts [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+      ],
+      description: 'Remove deprecated/empty smart contracts from CR Oracle'
+    }
+  ],
+  description: `
+  TIP-121c: veBAL OTC
+
+  This proposal sets up the protocol to be able to transfer its 112k veBAL to Aave Companies for OTC trade.
+
+  This proposal also includes several cleanup items to wrap up previous proposals.
```
Comment (eswak) [inline proposals/description/tip_vebalotc.ts]:
```suggestion
```

File: contracts/metagov/VeBalHelper.sol [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+    /// @notice Withdraw ERC20 tokens that are on the pcvDeposit
+    /// @dev this will be needed to withdraw B-80BAL-20WETH after exitLock(),
+    /// but also to withdraw BAL and bb-a-usd earned from protocol fees
+    function withdrawERC20(
+        address token,
+        address to,
+        uint256 amount
+    ) external onlyOwner {
+        require(token == ERC20_B_80BAL_20WETH || token == ERC20_BAL || token == ERC20_BB_A_USD, "INVALID_TOKEN");
```
Comment (eswak) [inline contracts/metagov/VeBalHelper.sol]: Probably needs to be removed (bbausd may not be used for platform fees in the future, etc)

File: proposals/description/tip_vebalotc.ts [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+  title: 'TIP-121c: veBAL OTC',
+  commands: [
+    // 1. Handle veBAL OTC
+    {
+      target: 'proxyAdmin',
+      values: '0',
+      method: 'changeProxyAdmin(address,address)',
+      arguments: (addresses) => [addresses.balancerGaugeStaker, addresses.aaveCompaniesMultisig],
+      description: 'Transfer proxy ownership of balancerGaugeStaker to aaveCompaniesMultisig'
+    },
```
Comment (eswak) [inline proposals/description/tip_vebalotc.ts]: Todo: upgrade implementation before proposal execution, to allow boost delegation

File: contracts/metagov/VeBalHelper.sol (hunk 9 above)
Comment (eswak) [reply contracts/metagov/VeBalHelper.sol]: Removed this restriction

File: contracts/metagov/VeBalHelper.sol [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+
+    // ----------------------------------------------------------------------------------
+    // Delegation Management
+    // ----------------------------------------------------------------------------------
+
+    /// @notice sets the snapshot delegate
+    function setDelegate(address newDelegate) external onlyOwner {
+        pcvDeposit.setDelegate(newDelegate);
+    }
+
```
Comment (miguelmtzinf) [inline contracts/metagov/VeBalHelper.sol:41]: MIssing a function to clear delegate here

Comment (miguelmtzinf) [reply contracts/metagov/VeBalHelper.sol:41]: thanks, lgtm!

File: protocol-configuration/mainnetAddresses.ts [hunk 12]
```diff
@@ -855,6 +885,19 @@ export const MainnetContractsConfig = {
...
   },
 
+  aaveCompaniesMultisig: {
+    artifactName: 'unknown',
+    address: '0xe8d4D93D9728bD673b0197673a230F62255C7846',
+    category: AddressCategory.External
+  },
+
+  // TODO: Update to correct multisig, and not Aave companies when ready
+  aaveCompaniesDaiEscrowMultisig: {
```
Comment (thomas-waite) [inline protocol-configuration/mainnetAddresses.ts:838]: Note this TODO. Important we update to be the correct escrow multisig - currently I've put in the standard Aave companies multisig so I can test the proposal.

**Review by miguelmtzinf — DISMISSED** [review]

**Comment by thomas-waite** [comment]
TODO: Check effect on PCV stats

**Comment by thomas-waite** [comment]
Remaining TODOs:
~~1. Investigate why this sends PCV equity negative - it's because we're optimistically removing 1M DAI from the backing. Maybe add a hardcode 1M NamedPCVDeposit to cover it?~~
~~2. Call `_initialise()` on the BalancerGaugeStakerImplV2 behaviour contract. Set owner to aave escrow multisig and set the voting escrow delegation~~
~~3. Withdraw 25k FEI from pool 8~~ NOT DOING

**Review by miguelmtzinf — DISMISSED** [review]

File: protocol-configuration/mainnetAddresses.ts (hunk 12 above)
Comment (thomas-waite) [reply protocol-configuration/mainnetAddresses.ts:838]: done

File: protocol-configuration/mainnetAddresses.ts [hunk 13]
```diff
@@ -855,6 +889,24 @@ export const MainnetContractsConfig = {
...
+    artifactName: 'unknown',
+    address: '0xA27A83A2433EbEB9f658B4420074B454846e1C5B',
+    category: AddressCategory.External
+  },
+
+  escrowedAaveDaiPCVDeposit: {
+    artifactName: 'ERC20PCVDepositWrapper',
+    address: '0x82c55A1Ab5C6F4b8e162b7dE24b50A38E1aFd38f',
+    category: AddressCategory.External
```
Comment (Joeysantoro) [inline protocol-configuration/mainnetAddresses.ts]: This is not external its PCV. While you are at it, can you move the Deprecated contracts down into the deprecated section

Comment (thomas-waite) [reply protocol-configuration/mainnetAddresses.ts]: done

File: protocol-configuration/mainnetAddresses.ts [hunk 14]
```diff
@@ -855,6 +823,24 @@ export const MainnetContractsConfig = {
...
+  },
+
+  aaveCompaniesDaiEscrowMultisig: {
+    artifactName: 'unknown',
+    address: '0xA27A83A2433EbEB9f658B4420074B454846e1C5B',
+    category: AddressCategory.External
+  },
+
+  escrowedAaveDaiPCVDeposit: {
```
Comment (Joeysantoro) [inline protocol-configuration/mainnetAddresses.ts]: please move this to the PCV section above

Comment (thomas-waite) [reply protocol-configuration/mainnetAddresses.ts]: done

**Comment by thomas-waite** [comment]
Calldata:
0x7d5e81e20000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000042000000000000000000000000000000000000000000000000000000000000007c00000000000000000000000000000000000000000000000000000000000001a20000000000000000000000000000000000000000000000000000000000000001c0000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b90000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b90000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b90000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b9000000000000000000000000f8c2b645988b7658e7748ba637fe25bdd46a704a00000000000000000000000066977ce30049cd0e443216bf26377966c3a109e2000000000000000000000000f8c2b645988b7658e7748ba637fe25bdd46a704a000000000000000000000000ff6f59333cfd8f4ebc14ad0a0e181a83e655d257000000000000000000000000ff6f59333cfd8f4ebc14ad0a0e181a83e655d2570000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b9000000000000000000000000956f47f50a910163d8bf957cf5846d573e7f87ca000000000000000000000000956f47f50a910163d8bf957cf5846d573e7f87ca0000000000000000000000007842186cdd11270c4af8c0a99a5e0589c7f249ce0000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b9000000000000000000000000f7991f4698ffb6716982aec7f78964dd731c4a54000000000000000000000000ff6f59333cfd8f4ebc14ad0a0e181a83e655d2570000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b90000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b90000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b90000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b90000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b90000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b90000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b90000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b90000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b9000000000000000000000000e0c7de94395b629860cbb3c42995f300f56e6d7a0000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b90000000000000000000000009cc46ab5a714f7cd24c59f33c5769039b5872491000000000000000000000000000000000000000000000000000000000000001c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001c000000000000000000000000000000000000000000000000000000000000038000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000480000000000000000000000000000000000000000000000000000000000000050000000000000000000000000000000000000000000000000000000000000005800000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000068000000000000000000000000000000000000000000000000000000000000007000000000000000000000000000000000000000000000000000000000000000760000000000000000000000000000000000000000000000000000000000000092000000000000000000000000000000000000000000000000000000000000009800000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000a800000000000000000000000000000000000000000000000000000000000000b200000000000000000000000000000000000000000000000000000000000000b800000000000000000000000000000000000000000000000000000000000000bc00000000000000000000000000000000000000000000000000000000000000c400000000000000000000000000000000000000000000000000000000000000cc00000000000000000000000000000000000000000000000000000000000000d400000000000000000000000000000000000000000000000000000000000000dc00000000000000000000000000000000000000000000000000000000000000e400000000000000000000000000000000000000000000000000000000000000ec00000000000000000000000000000000000000000000000000000000000000f400000000000000000000000000000000000000000000000000000000000000fc0000000000000000000000000000000000000000000000000000000000000104000000000000000000000000000000000000000000000000000000000000010c0000000000000000000000000000000000000000000000000000000000000114000000000000000000000000000000000000000000000000000000000000011c000000000000000000000000000000000000000000000000000000000000000442f2ff15d0866eae1216ed05a11636a648003f3f62921eb97ccb05acc30636f62958a8bd6000000000000000000000000b02ce4d72124b98df4eab4184467d7da0023f9fb0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000442f2ff15da100760f521bbb2848bef0b72ea29301f6a6b0605d004243f0eea2b1c359f7c7000000000000000000000000b02ce4d72124b98df4eab4184467d7da0023f9fb0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000442f2ff15d3bee38c33241595abfefa470fd75bfa1cc9cb01eff02cf6732fd2baea4ea4241000000000000000000000000b02ce4d72124b98df4eab4184467d7da0023f9fb0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000442f2ff15db02f76effb323167cad756bb4f3edbfb9d9291f9bfcdc72c9ceea005562f32eb000000000000000000000000b02ce4d72124b98df4eab4184467d7da0023f9fb00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004499a88ec400000000000000000000000066977ce30049cd0e443216bf26377966c3a109e2000000000000000000000000bf3f6c7821e4d7d5df204dea4871b2e8d68c1ccd000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044e5c9c2ed000000000000000000000000b02ce4d72124b98df4eab4184467d7da0023f9fb000000000000000000000000b496ff44746a8693a060fafd984da41b253f67900000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000447eff275e00000000000000000000000066977ce30049cd0e443216bf26377966c3a109e2000000000000000000000000e8d4d93d9728bd673b0197673a230f62255c7846000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000024dea0289200000000000000000000000082c55a1ab5c6f4b8e162b7de24b50a38e1afd38f000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000184e0baee2e0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000008cba3149b95084a61bbab9e01110b0fb92c9a289000000000000000000000000d8eb546726d449fc1ded06dfeca800a2fa8bb93000000000000000000000000066977ce30049cd0e443216bf26377966c3a109e2000000000000000000000000cd1ac0014e2ebd972f40f24df1694e6f528b2fd4000000000000000000000000df9ff5c077d9f3427ade67ac2d27a864be6f3187000000000000000000000000f24401f6992faeacbc5d6c6991db15b5f8364a1b0000000000000000000000004378de2f2991fbed6616b34ac7727e7653713712000000000000000000000000c6d675ca5217d39c3a5e366141060fc2d1ea2b820000000000000000000000006e5f2745c08249a190239763706473be0b72816d0000000000000000000000008ffae111ab06f532a18418190129373d14570014000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000024261707fa000000000000000000000000d51dba7a94e1adea403553a8235c302cebf41a3c00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004440c10f19000000000000000000000000d51dba7a94e1adea403553a8235c302cebf41a3c00000000000000000000000000000000000000000000d3c21bcecceda1000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044095ea7b30000000000000000000000007842186cdd11270c4af8c0a99a5e0589c7f249ce00000000000000000000000000000000000000000000d3c21bcecceda10000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000642b83cccd000000000000000000000000a27a83a2433ebeb9f658b4420074b454846e1c5b00000000000000000000000000000000000000000000d3c21bcecceda10000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000024cfbd4885000000000000000000000000d51dba7a94e1adea403553a8235c302cebf41a3c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000048456cb590000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000445c38eb3a0000000000000000000000006b175474e89094c44da98b954eedeac495271d0f0000000000000000000000002374800337c6be8b935f96aa6c10b33f9f12bd40000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044d547741fb02f76effb323167cad756bb4f3edbfb9d9291f9bfcdc72c9ceea005562f32eb000000000000000000000000d51dba7a94e1adea403553a8235c302cebf41a3c000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044d547741fa100760f521bbb2848bef0b72ea29301f6a6b0605d004243f0eea2b1c359f7c7000000000000000000000000d51dba7a94e1adea403553a8235c302cebf41a3c000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044d547741f3bee38c33241595abfefa470fd75bfa1cc9cb01eff02cf6732fd2baea4ea4241000000000000000000000000d51dba7a94e1adea403553a8235c302cebf41a3c000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044d547741f46797b318ce8c2d83979760ef100a5c0fdb980de4b574d6142ce4d0afce307ed000000000000000000000000d51dba7a94e1adea403553a8235c302cebf41a3c000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044d547741f55435dd261a4b9b3364963f7738a7a662ad9c84396d64be3365284bb7f0a5041000000000000000000000000c297705acf50134d256187c754b92fa37826c019000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044d547741f55435dd261a4b9b3364963f7738a7a662ad9c84396d64be3365284bb7f0a504100000000000000000000000002435948f84d7465fb71de45aba6098fc6ec2993000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044d547741f0866eae1216ed05a11636a648003f3f62921eb97ccb05acc30636f62958a8bd600000000000000000000000002435948f84d7465fb71de45aba6098fc6ec2993000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044d547741f29af6c210963c1cf458c6a5bf082996cf54b23ebba0c0fb8ae110e8e43371c710000000000000000000000006c7af43ce97686e0c8acbbc03b2e4f313c0394c7000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044d547741f2172861495e7b85edac73e3cd5fbb42dd675baadf627720e687bcfdaca025096000000000000000000000000d51dba7a94e1adea403553a8235c302cebf41a3c000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044d547741ffd643c72710c63c0180259aba6b2d05451e3591a24e58b62239378085726f783000000000000000000000000dde8aa537c5b289de9cede462e6f0ec3a3a99e39000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044d547741f0866eae1216ed05a11636a648003f3f62921eb97ccb05acc30636f62958a8bd6000000000000000000000000221fff24fb66da3c722c7c5b856956a6a30c0179000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044f3fef3a30000000000000000000000004d9629e80118082b939e3d59e69c82a2ec08b4d5000000000000000000000000000000000000000000000541ec5c5a0e56995bb90000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000003f35449502d313231633a20766542414c204f54432077697468204161766520436f6d70616e6965730a0a20205449502d313231633a20766542414c204f54432077697468204161766520436f6d70616e6965730a0a2020546869732070726f706f73616c2073657473207570207468652070726f746f636f6c20746f207472616e73666572203131326b20766542414c20746f204161766520436f6d70616e69657320666f72204f54432074726164652e0a0a4161766520436f6d70616e6965732077696c6c2073656e6420312c3030302c3030302044414920746f20626520657363726f776564206f6e206120322f33206d756c74697369672077697468204161766520436f6d70616e6965732c2042616c616e6365722044414f2c20616e642054726962652044414f20287769746820657377616b2e657468206173207468652064657369676e6174656420726570726573656e746174697665292e0a0a496e206164646974696f6e2c207468652070726f706f73616c206d6f76657320314d204441492066726f6d20746865204645492050534d20746f2074686520657363726f77206d756c746973696720746f2062652073656e7420746f207468652054524942452052656465656d65722e0a0a416674657220746865204161766520436f6d70616e696573207465616d20636f6e6669726d7320746865792063616e2070726f7065726c79207573652074686520766542414c2d686f6c64696e6720736d61727420636f6e7472616374732c2074686520322f33206d756c74697369672077696c6c207472616e7366657220314d2044414920746f207468652054524942452052656465656d65722c20676976696e6720545249424520686f6c646572732061636365737320746f20746865206f7074696d616c20616d6f756e74206f66206578636573732050435620666f7220726564656d7074696f6e732e0a0a4166746572207468652044414f20566f746520746f2064697361626c6520676f7665726e616e63652c207468652066696e616c20314d204441492066726f6d204161766520436f6d70616e69657320776f756c642062652073656e74206261636b20746f20746865204645492050534d2066726f6d2074686520e28594204d756c74697369672e0a0a4c6173746c792c207468652070726f706f73616c206675727468657220636c65616e7320757020756e7573656420726f6c657320696e207468652073797374656d20616e6420706572666f726d73206d61696e74656e616e6365206f6e2074686520636f6c6c61746572616c697a6174696f6e206f7261636c6520736f207468617420697420697320696e20746865206d6f737420757020746f20646174652073746174652e0a0a202000000000000000000000000000

**Comment by thomas-waite** [comment]
Screenshot 2022-10-05 at 16 29 15
PCV stats after proposal

**Review by Joeysantoro — APPROVED** [review]

**Pull request merged**